feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/README.md

```diff
@@ -2,53 +2,141 @@
 
 ## Summary
 
-BGP Graceful Restart
+This is to test the BGP graceful restart capability for BGP. A router that supports BGP graceful restart can work either as a Restarting speaker mode or in helper mode. By advertising BGP graceful restart capability, a router announces to the peer its ability to,
+1.  [Restarting Speaker] Maintain forwarding state on all the routes in its FIB even when its BGP process is restarting. Therefore the peer functioning as a helper should continue to direct flows at the subject router undergoing BGP process restart.
+2.  [Helper Router] Support a peer whose BGP process is restarting by continuing to direct flows at the peer.
+
+While testing for the above, this test also confirms that the implementation respects stale-routes-time timer setting.
+
+
+## Topology
+Create the following connections:
+```mermaid
+graph LR; 
+A[ATE:Port1] -- EBGP --> B[Port1:DUT:Port2];
+B -- IBGP --> C[Port2:ATE];
+```
 
 ## Procedure
 
-*   Establish eBGP sessions between:
-    *   ATE port-1 and DUT port-1
-    *   ATE port-2 and DUT port-2
-    *   Configure allow route-policy under BGP peer-group address-family
+**RT-1.4.1: Enable and validate BGP Graceful restart feature**
+*   Configure EBGP peering between ATE:Port1 and DUT:Port1
+*   Configure IBGP peering between ATE:Port2 and DUT:Port2
+*   Ensure that the EBGP and IBGP peering are setup for IPv4-Unicast and IPv6-unicast AFI-SAFIs. Total 2xpeer-groups (1 per protocol) with 1 BGP session each.  
+*   Enable `Graceful-Restart` capability at the `Peer-Group` level.
+*   Ensure that the `restart-time` and the `stale-routes-time` are configured at the `Global` level
```

Comment:
what should stale-routes-time be set to?  (performance target for bgp restart)

Reply:
I have defined the numbers for RFC4724 v/s RFC8538